packages/x-components/src/utils/__tests__/html.spec.ts

```diff
@@ -24,3 +24,78 @@ describe(`testing ${isElementEqualOrContained.name} utility method`, () => {
     expect(isElementEqualOrContained(a, b)).toBe(false);
   });
 });
+
+describe(`testing ${getTargetElement.name} utility method`, () => {
```

Comment:
Because this is a browser behavior,  I would move this test to Cypress Component testing, rather than using `jest` and `jsdom`. Then we can add more browsers and test that this works properly in all of them.

Reply:
I tried this in Cypress and it gives me much more confidence than `jsdom` 🤣.
```
import { getTargetElement } from '../../src/utils/html';

describe(`testing ${getTargetElement.name} utility method`, () => {
  beforeEach(() => {
    document.body.innerHTML = '';
  });

  function createBasicDOMTree(): [HTMLDivElement, HTMLButtonElement] {
    const container = document.createElement('div');
    const button = container.appendChild(document.createElement('button'));
    button.textContent = 'Click';

    return [container, button];
  }

  function createShadowDOMTree(mode: 'open' | 'closed'): [HTMLDivElement, HTMLButtonElement] {
    const [container, button] = createBasicDOMTree();

    const shadowContainer = document.createElement('div');
    const shadowRoot = shadowContainer.attachShadow({ mode });
    shadowRoot.appendChild(container);

    return [shadowContainer, button];
  }

  it('returns the element that triggered the event', () => {
    const [container, button] = createBasicDOMTree();
    document.body.appendChild(container);
    const spy = cy.spy(getTargetElement);
    container.addEventListener('click', spy);
    button.click();
    cy.wrap(spy).should('have.callCount', 1);
    cy.wrap(spy).should('have.returned', button);
  });

  it('returns the element that triggered the event in a opened shadow DOM context', () => {
    const [shadowContainer, button] = createShadowDOMTree('open');
    const spy = cy.spy(getTargetElement);
    shadowContainer.addEventListener('click', spy);
    document.body.appendChild(shadowContainer);
    button.click();
    cy.wrap(spy).should('have.callCount', 1);
    cy.wrap(spy).should('have.returned', button);
  });

  it("doesn't return the element that triggered the event in a closed shadow DOM context", () => {
    const [shadowContainer, button] = createShadowDOMTree('closed');
    const spy = cy.spy(getTargetElement);
    shadowContainer.addEventListener('click', spy);
    document.body.appendChild(shadowContainer);
    button.click();
    cy.wrap(spy).should('have.callCount', 1);
    cy.wrap(spy).should('have.returned', shadowContainer);
  });
});
```

Reply:
I agree with you. Migrated unit test to Cypress